### DISCUSSION TITLE
Session Progress Bar

### DISCUSSION THREAD

**Discussion by wkaisertexas** [focustimerhq/FocusTimer#599] [Ideas]
I am currently in the midst of a lifestyle change from Windows to Linux. As a High School student, I use the pomodoro technique. This app is almost perfect, but there are some QOL features that I think we should implement from the Windows pomodoro software I used previously.

![image](https://user-images.githubusercontent.com/27795014/141310543-1420cf8e-c258-4cb5-b490-3fcbd0487900.jpeg)

This is Focus 10, and the progress bar on top of the circle is something that I missed. It shows where you are out of your 4 pomodoro schedule.

The option to reset your 4 pomodoro schedule (the refresh button in the bottom right corner) would be nice.

**Comment by sojusnik** [comment]
This bar looks like a very helpful feature! Would love to see it too!

**Comment by wkaisertexas** [comment]
Also, is there a particular reason why sessions are started automatically. Could there be an option in preferences to manually start the next pomodoro?

**Comment by kamilprusko** [comment]
I intend to add it while porting to gtk4. Mockup:

![g1311](https://user-images.githubusercontent.com/1135423/141317310-a6e99f50-f7e0-4aae-a4f4-fde3b9501e3f.png)

Any suggestions where to put the option to reset session cycle? Current idea is to auto reset after 1h of inactivity, with an in-app notification to undo it.

> Also, is there a particular reason why sessions are started automatically. Could there be an option in preferences to manually start the next pomodoro?

In GNOME it waits for user activity before starting next pomodoro. On desktops where this behavior is unavailable, option to manually start pomodoro sounds reasonable.

**Reply by wkaisertexas** [reply]
### Additional Feature
The ability to skip to the end of of a pomodoro. Though not exactly in line with the pomodoro technique, it is useful if you finished a task with a minute or so left, and you want to take a break before starting the next one. 

### Button Placement
You could have the rest button on the left and the skip button on the right on the bottom.

**Comment by kamilprusko** [comment]
It landed into `main` branch. Flatpak bundle for testing is [here](https://github.com/gnome-pomodoro/gnome-pomodoro/discussions/742)
